scripts/generate_coldcard_hsm_policy.js

```diff
@@ -0,0 +1,133 @@
+require('dotenv').config({
+    override: true
+});
+
+const fs = require('fs');
+const WEEK_IN_MINUTES = 1440 * 7; // Seven days in minutes
+
+// Reference: https://coldcard.com/docs/hsm/rules/#per-period-limit
+// get period: (integer) velocity period, in minutes, shared across all the rules
+const getPeriod = () => {
+    try {
+        const period = parseInt(process.argv[2]);
+        console.log('Using custom period for HSM policy in minutes:', period);
+        return period;
+    } catch (error) {
+        console.log('Using default period for HSM policy in minutes:', WEEK_IN_MINUTES);
+        return WEEK_IN_MINUTES;
+    }
+};
+
+const split_address = (env_var) => {
+    if (!env_var) return [];
+    try {
+        return env_var.split(' ').map((address_item) => {
+            const [description, address] = address_item.split(':');
+            return {
+                address,
+                description
+            };
+        });
+    } catch (error) {
+        console.log('Error splitting addresses:', error);
+    }
+    return [];
+};
+
+const get_whitelist = () => {
+    const tag_by_address = split_address(process.env.TAG_BY_ADDRESS);
+    const address_book = split_address(process.env.ADDRESS_BOOK);
+    const exchange_addresses = [
+        'KRAKEN',
+        'COINBASE',
+        'COINBASE_EXCHANGE',
+        'COINBASE_PRIME',
+        'GEMINI',
+        'BFX',
+        'BINANCE',
+        'OKX',
+        'BYBIT',
+        'KUCOIN'
+    ].reduce((acc, curr) => {
+        const withdrawal_address_key = `${curr}_WITHDRAWAL_ADDRESS`;
```

Comment:
I don't think withdrawal addresses should be on the whitelist - only the deposit addresses